pkg/ck8s/workload_cluster.go

```diff
@@ -209,13 +208,28 @@ func (w *Workload) GetCertificatesExpiryDate(ctx context.Context, machine *clust
 	return response.ExpiryDate, nil
 }
 
-func (w *Workload) RefreshCertificates(ctx context.Context, machine *clusterv1.Machine, nodeToken string, expirationSeconds int, extraSANs []string) (int, error) {
+func (w *Workload) ApproveCertificates(ctx context.Context, machine *clusterv1.Machine, seed int) error {
+	request := apiv1.ClusterAPIApproveWorkerCSRRequest{}
+	response := &apiv1.ClusterAPIApproveWorkerCSRResponse{}
+	k8sdProxy, err := w.GetK8sdProxyForControlPlane(ctx, k8sdProxyOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to create k8sd proxy: %w", err)
+	}
+
+	header := w.newHeaderWithCAPIAuthToken()
+
+	if err := w.doK8sdRequest(ctx, k8sdProxy, http.MethodPost, "1.0/x/capi/refresh-certs/approve", header, request, response); err != nil {
```

Comment:
Maybe we can use the `k8s-snap-api` endpoint here? Something like `url.JoinPath(apiv1.K8sdAPIVersion, apiv1.ClusterAPICertificatesApproveRPC)`

Reply:
Let's create a follow up PR to refactor the URLs across the codebase. It would be difficult to introduce this type of change in some parts while leaving others unchanged. What do you think?

Reply:
Of course, let's go!